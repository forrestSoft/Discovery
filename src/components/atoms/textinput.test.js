import React from 'react'
import {shallow, mount} from 'enzyme'
import shallowToJson from 'enzyme-to-json'

import TextInput from './textinput'

it('render matches snapshot', () => {
  const textinput = shallowToJson(shallow(
  	<TextInput 
      placeholder="I am the placeholder"
    />
  )) 
  expect(textinput).toMatchSnapshot();
});

it('render with value matches snapshot', () => {
  const textinput = shallowToJson(shallow(
  	<TextInput 
      placeholder="I am the placeholder"
      value="I am the value"
    />
  )) 
  expect(textinput).toMatchSnapshot();
});

it('render invalid matches snapshot', () => {
  const textinput = shallowToJson(shallow(
  	<TextInput 
      placeholder="I am the placeholder"
      valid={false}
    />
  )) 
  expect(textinput).toMatchSnapshot();
});

it('render valid matches snapshot', () => {
  const textinput = shallowToJson(shallow(
  	<TextInput 
      placeholder="I am the placeholder"
      valid={true}
    />
  )) 
  expect(textinput).toMatchSnapshot();
});

it('calls passed onChange on "onChange"', () => {
	const mock = jest.fn().mockReturnValue('onClick')
	const textinput = mount(
  	<TextInput 
  		placeholder={'I am a checkbox'}
  		onChange={mock}
  	/>
  )

	textinput.find('input[type="text"]').simulate('change')
	expect(mock).toBeCalled()
})
