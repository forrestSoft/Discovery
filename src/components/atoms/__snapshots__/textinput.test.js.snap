// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render invalid matches snapshot 1`] = `
<input
  className="invalid"
  onChange={null}
  placeholder="I am the placeholder"
  type="text"
/>
`;

exports[`render matches snapshot 1`] = `
<input
  className="invalid"
  onChange={null}
  placeholder="I am the placeholder"
  type="text"
/>
`;

exports[`render valid matches snapshot 1`] = `
<input
  className="valid"
  onChange={null}
  placeholder="I am the placeholder"
  type="text"
/>
`;

exports[`render with value matches snapshot 1`] = `
<input
  className="invalid"
  onChange={null}
  placeholder="I am the placeholder"
  type="text"
  value="I am the value"
/>
`;
